src/common/security/v2token/context.go

```diff
@@ -5,22 +5,22 @@ import (
 	"strings"
 
 	registry_token "github.com/docker/distribution/registry/auth/token"
-	"github.com/goharbor/harbor/src/common/models"
 	"github.com/goharbor/harbor/src/common/rbac"
 	"github.com/goharbor/harbor/src/common/security"
 	"github.com/goharbor/harbor/src/lib/log"
 	"github.com/goharbor/harbor/src/pkg/permission/types"
 	"github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/pkg/project/models"
 )
 
 // tokenSecurityCtx is used for check permission of an internal signed token.
 // The intention for this guy is only for support CLI push/pull.  It should not be used in other scenario without careful review
 // Each request should have a different instance of tokenSecurityCtx
 type tokenSecurityCtx struct {
-	logger    *log.Logger
-	name      string
-	accessMap map[string]map[types.Action]struct{}
-	pm        project.Manager
+	logger     *log.Logger
+	name       string
+	accessMap  map[string]map[types.Action]struct{}
+	getProject func(int64) (*models.Project, error)
```

Comment:
Why don't use the project manager directly?

Reply:
project manager needs ctx as its first argument but the `Can` method can't provide ctx argument so use a function here and the ctx argument from the ctx when create v2token security.